core/src/mender-client.c

```diff
@@ -784,6 +789,20 @@ mender_client_initialization_work_function(void) {
             goto REBOOT;
         }
         free(storage_deployment_data);
+
+        /* Check deployment data version compatibility */
```

Comment:
> 1. It is the same situation with the same action -> can we move this check together with the rest?

Yes there are some checks in `mender_client_authentication_work_function` but it actually does not reboot if those checks fail. It just looks like it skips publishing deployment status. I'm not really sure what it is supposed to do.

> 2. Pre-existing, but this can be a problem if we don't have a way to tell that we are restarting all the time. For this we would need somehow to detect that this is the problem and wipe out the state data. I'll think about this of of the PR.

Currently it looks like the deployment data is wiped immediately in `mender_client_initialization_work_function` function, just before reboot. So this would not be an issue as far as I can tell. But there probably should be some time to live variable in it.